### DISCUSSION TITLE
Glob pattern support in FileSystem tab search bar

### DISCUSSION THREAD

**Discussion by tilkinsc** [godotengine/godot-proposals#8405] [Editor]
Should FileSystem's search support glob patterns. For example, ico_*.{png,jpg,jpeg} would potentially resolve to a naming convention where all icon files are prefixed with ico_ and the file type could be one of png jpg or jpeg.

![image](https://github.com/godotengine/godot-proposals/assets/7494772/1cb2ddac-79ea-450e-9fd0-11dc42f1bf34)

**Comment by Calinou** [comment]
Godot has support for `*` and `?` glob syntax with its [`String.match()`](https://github.com/godotengine/godot/blob/bdd9034ad05e1824ff5d9c750acd87caeafe6dca/doc/classes/String.xml#L559) method, but not `{foo,bar}`.

**Reply by tilkinsc** [reply]
![image](https://github.com/godotengine/godot-proposals/assets/7494772/be305ac4-6220-4ecc-8574-ef7130c32271)

**Reply by Calinou** [reply]
What I meant is that the `String.match()` method supported some of the globbing you requested (but not all), not that the FileSystem dock has support for using globs :slightly_smiling_face: 

I think this makes sense to implement, but this needs to be done consistently with the Scene tree dock and inspector filter bars to have a consistent UX.

**Reply by Mickeon** [reply]
Bit of an off-hand note about the latter. Node names can contain `*` `?` just fine, which has you wonder how this would work in practice in the Scene tree dock...

**Reply by tilkinsc** [reply]
Good call out. I think * and ? should be invalid characters as they aren't a name they are symbols.

**Reply by Calinou** [reply]
Changing the valid characters in node names is a compatibility-breaking change. I'd just make it so using `\*` or `\?` escapes the character for search (this may already be the case in `String.match()`).
